File: GameFramePro/Assets/.lpproj/Apis/Unity/CS/GameFramePro/NetWorkEx/AudioClipDownloadManager.lua
---@class CS.GameFramePro.NetWorkEx.AudioClipDownloadManager : CS.GameFramePro.NetWorkEx.UnityWebRequestDownLoadManager
CS.GameFramePro.NetWorkEx.AudioClipDownloadManager = {}

---@property readonly CS.GameFramePro.NetWorkEx.AudioClipDownloadManager.S_Instance : CS.GameFramePro.NetWorkEx.AudioClipDownloadManager
CS.GameFramePro.NetWorkEx.AudioClipDownloadManager.S_Instance = nil

---@return CS.GameFramePro.NetWorkEx.AudioClipDownloadManager
function CS.GameFramePro.NetWorkEx.AudioClipDownloadManager()
end

---@param taskUrl : CS.System.String
---@param type : CS.UnityEngine.AudioType
---@param callback : CS.System.Action
---@param priorityEnum : CS.GameFramePro.NetWorkEx.UnityTaskPriorityEnum
function CS.GameFramePro.NetWorkEx.AudioClipDownloadManager:GetDataFromUrl(taskUrl, type, callback, priorityEnum)
end

File: GameFramePro/GameFrame_Client/Assets/Scripts/Core/Module/NetWork/Download/Downloader/UnityWebRequest/AudioClipDownloadManager.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;
using System;
using UnityEngine.Networking;

namespace GameFramePro.NetWorkEx
{
    /// <summary>/// 声音资源下载器/// </summary>
    public class AudioClipDownloadManager : BaseDownloadManager<UnityWebRequestDownloadTask, UnityWebRequest>
    {
        protected static AudioClipDownloadManager s_Instance = null;

        public static AudioClipDownloadManager S_Instance
        {
            get
            {
                if (s_Instance == null)
                {
                    s_Instance = new AudioClipDownloadManager();
                    DownloadManager.RegisterDownloadManager(s_Instance);
                }
                return s_Instance;
            }
        } // 

        #region  各种类型获取数据的接口

        /// <summary>/// 获取一个数据/// </summary>
        public virtual UnityWebRequestDownloadTask GetDataFromUrl(string taskUrl, AudioType type, Action<UnityWebRequest, bool, string> callback, TaskPriorityEnum priorityEnum = TaskPriorityEnum.Normal)
        {
            if (string.IsNullOrEmpty(taskUrl))
            {
                Debug.LogError("TryAddToDownloadTaskLink Fail,Parameter is null of taskUrl");
                return null;
            }
            //TODO从下载的缓存中读取数据

            //将新的下载任务添加到缓存的下载队列中
            UnityWebRequestDownloadTask newDownLoadTask = UnityWebRequestDownloadTask.GetUnityWebRequestDownloadTask(taskUrl, callback, priorityEnum);
            newDownLoadTask.DownloadTaskCallbackData.downloadHandler = new DownloadHandlerAudioClip(taskUrl, AudioType.MPEG);
            newDownLoadTask.ChangeDownloadState(TaskStateEum.Initialed);
            AllCacheDownLoadTasks.Add(newDownLoadTask);
            return newDownLoadTask;
        }


      

        #endregion
    }
}
